### PR TITLE
api400 model and algo

### PR DESCRIPTION
**Pull request by ankitenasaxena**
- api400 model file
- apicl sort dependency algo

### CODE REVIEW

File: frontend/apps/monastery/views/api400-designer/model/algos.mjs [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+    }
+};
+
+/**
+ * Fecth the node which was included in the dependencies of other node
+ * @param {*} node Command Object
+ * @param {*} allnodes All the Commmand's Object
+ * @returns found node object
+ */
+const checkNodeInAllNodes = function (node, allnodes) {
```
Comment (TekMonksGitHub) [inline frontend/apps/monastery/views/api400-designer/model/algos.mjs:535]: If this is internal then add _ at the start, else move up to public functions. Don't mix non-public and public functions.

File: frontend/apps/monastery/views/api400-designer/model/api400model.mjs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    blackboard.registerListener(MSG_NODE_DESCRIPTION_CHANGED, message => nodeDescriptionChanged(message.nodeName,
+        message.id, message.description));
+    blackboard.registerListener(MSG_ARE_NODES_CONNECTABLE, message => isConnectable(message.sourceName,
+        message.targetName, message.sourceID, message.targetID), true);
+    blackboard.registerListener(MSG_GET_MODEL, message => getModelAsFile(message.name), true);
+    blackboard.registerListener(MSG_RESET, _ => { api400modelObj = EMPTY_MODEL, idCache = {}, current_command_bundle = DEFAULT_BUNDLE; }, true);
+    blackboard.registerListener(MSG_LOAD_MODEL, message => loadModel(message.data));
+}
+
+function loadModel(jsonModel) {
```
Comment (TekMonksGitHub) [inline frontend/apps/monastery/views/api400-designer/model/api400model.mjs:32]: I still do NOT understand how this reads APICL. Seems the back parser is simply not written!

File: frontend/apps/monastery/views/api400-designer/model/algos.mjs (hunk 1 above)
Comment (ankit-saxena-deeplogictech) [reply frontend/apps/monastery/views/api400-designer/model/algos.mjs:535]: We are callling/using this function from 2 different files, that is why we have not put the _

Comment (TekMonksGitHub) [reply frontend/apps/monastery/views/api400-designer/model/algos.mjs:535]: Then move it amongst public functions - the org of code should be

- imports / requires
- constants
- public functions
- private functions
- export declaration at the end

File: frontend/apps/monastery/views/api400-designer/model/api400model.mjs (hunk 2 above)
Comment (ankit-saxena-deeplogictech) [reply frontend/apps/monastery/views/api400-designer/model/api400model.mjs:32]: We are calling the APICL parser funtion at the time of Upload it from local OR get it from server , in Open.mjs file.

We havn't this file for this release-1 PR , we will push it in release-2 today.

Comment (TekMonksGitHub) [reply frontend/apps/monastery/views/api400-designer/model/api400model.mjs:32]: OK

File: frontend/apps/monastery/views/api400-designer/model/algos.mjs (hunk 1 above)
Comment (ankit-saxena-deeplogictech) [reply frontend/apps/monastery/views/api400-designer/model/algos.mjs:535]: Ok , we will change it. Thanks

**Pull request merged**